lib/saluki-components/src/destinations/datadog_metrics/endpoint/endpoints.rs

```diff
@@ -113,16 +116,42 @@ impl ResolvedEndpoint {
         Ok(Self {
             endpoint,
             api_key: api_key.to_string(),
+            config: None,
         })
     }
 
+    /// Creates a new  `ResolvedEndpoint` instance from an existing `ResolvedEndpoint`, adding an optional `RefreshableConfiguration`.
+    pub fn with_refreshable_configuration(self, config: Option<RefreshableConfiguration>) -> Self {
+        Self {
+            endpoint: self.endpoint,
+            api_key: self.api_key,
+            config,
+        }
+    }
+
     /// Returns the endpoint of the resolver.
     pub fn endpoint(&self) -> &Url {
         &self.endpoint
     }
 
     /// Returns the API key associated with the endpoint.
-    pub fn api_key(&self) -> &str {
+    pub fn api_key(&mut self) -> &str {
+        if let Some(config) = &self.config {
+            match config.try_get_typed::<String>("api_key") {
+                Ok(Some(api_key)) => {
+                    if !api_key.is_empty() && self.api_key != api_key {
+                        debug!("Refreshing api key.");
```

Comment:
Just some capitalization stuff and adding the endpoint so we know who got the refreshed API key.

```suggestion
                        debug!(endpoint = %self.endpoint, "Refreshed API key.");
```